### ISSUE TITLE
Public members in \Mpdf\Mpdf

### ISSUE THREAD

**Issue by KacerCZ** [mpdf/mpdf#502]
Class `\Mpdf\Mpdf` contains huge amount of public members.
Properties are declared using `var` and methods don't have accessibility specified making them effectively `public`. Some of methods have name starting with underscore probably indicating that they are meant to be `private`.

Release of new major version is good opportunity to correct this, but given amount of changes already present in 7.0 this seems it should be done in 8.0.

Is there any plan to change this and clean-up class interface?

**Comment by finwe** [comment]
Yes, it should be done. However triaging the properties turned out to be very hard.

I have a vague plan how to do this (creating a few value objects that would be shared among service classes, some public, some private), but I have not yet come up with a way to do this properly.

There are more things to consider - especially changes to the current "settings" approach via direct property modification.

With this change I would also like to unify the properties to a single camelCase format.

Much work to be done there.
